### ISSUE TITLE
[ValueCard] Enhancements for card

### ISSUE THREAD

**Issue by cgirani** [carbon-design-system/carbon-addons-iot-react#3357]
### What package is this for?

- [x] React
- [ ] Angular

### Summary

We need the ValueCard component to support:
1. Link under text
2. Icon next to the main text
3. _If possible_ extra section depending on card size

![image](https://user-images.githubusercontent.com/22034430/156229750-48abb838-40e8-451c-9ba4-1bfa2e0e6133.png)

![image](https://user-images.githubusercontent.com/22034430/156229778-05df23b7-6bd5-4b3f-a717-7cca8aff6e80.png)

### Specific timeline issues / requests

Do you want this work within a specific time period? Yes, we need this before end of May.
Is it related to an upcoming release? Yes, MAS Core 8.8

### Available extra resources

What resources do you have to assist this effort? Yes, I can work on it.

**Comment by davidicus** [comment]
@cgirani I think it should be pretty easy to perhaps leverage the existing trend text prop to achieve the links. For the combination of value card and list card in the bottom example, I think this is a perfect example of the need to have these different card types as composable pieces. If we had a value component, we could compose it easily with a custom card with a list component.

**Comment by davidicus** [comment]
@cgirani  This seems like a trivial list variation which makes me wonder if design should be validating the need for the change to existing list. @dianatran18 Do you have any input on this?

**Comment by dianatran18** [comment]
Hey @davidicus what is shown in the original screenshot aligns with the [read-only value](https://pages.github.ibm.com/ai-applications/design/components/read-only-value/usage) we have, which is a label + value pairing. We've seen teams with various padding and alignment, so we created this component documentation to create that consistency. 

If we can create that as a standalone component that can be used inside varying components and use cases, then that would be great. The contribution issue we opened for that is #3124
